File: Java/calcularAreaRectangulo/src/Teclado.java
import java.util.Scanner;

// Clase que permite leer datos de la consola
public class Teclado {
    // Scanner para leer datos de la consola
    private static final Scanner scanner = new Scanner(System.in);

    // Métodos para leer datos de la consola
    // Método para leer un número decimal.
    public static double readDouble() {
        // Variable para almacenar el número
        double numero;
        // Ciclo para leer el número, si no es un número válido se muestra un mensaje de error
        while (true) {
            try {
                numero = Double.parseDouble(scanner.nextLine());
                break;
            } catch (NumberFormatException e) {
                System.out.println("Entrada inválida. Intente de nuevo:");
            }
        }
        return numero;
    }

    // Método para leer un número decimal positivo.
    public static double readDoublePositive(Scanner scanner, String mensaje) {
        // Variable para almacenar el número
        double numero;
        // Ciclo para leer el número, si no es un número válido se muestra un mensaje de error
        while (true) {
            System.out.println(mensaje);
            try {
                numero = Double.parseDouble(scanner.nextLine());
                if (numero > 0) {
                    return numero;
                }
                System.out.println("El número debe ser mayor que 0. Intente de nuevo:");
            } catch (NumberFormatException e) {
                System.out.println("Entrada inválida. Intente de nuevo:");
            }
        }
    }

    // Método para leer un número entero.
    public static int readInteger() {
        // Variable para almacenar el número
        int numero;
        // Ciclo para leer el número, si no es un número válido se muestra un mensaje de error
        while (true) {
            try {
                numero = Integer.parseInt(scanner.nextLine());
                break;
            } catch (NumberFormatException e) {
                System.out.println("Entrada inválida. Intente de nuevo:");
            }
        }
        return numero;
    }

    // Método para leer un número entero positivo.
    public static String readText() {
        // Variable para almacenar el texto
        String texto;
        // Ciclo para leer el texto, si no es un texto válido se muestra un mensaje de error
        while (true) {
            texto = scanner.nextLine().trim();
            if (!texto.isEmpty()) {
                break;
            }
            System.out.println("El texto no puede estar vacío. Intente de nuevo:");
        }
        return texto;
    }
}

File: C#/CalcularAreaRectangulo/Teclado.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CalcularAreaRectangulo
{
    // Clase para leer datos desde la consola
    internal class Teclado
    {
        // Leer un número real
        public static double ReadDouble()
        {
            // Leer una línea de la consola
            double numero;
            // Mientras no se pueda convertir a double, seguir pidiendo el número
            while (!double.TryParse(Console.ReadLine(), out numero))
            {
                Console.WriteLine("Entrada inválida. Intente de nuevo:");
            }
            return numero;
        }
        // Leer un número real positivo
        public static double ReadDoublePositive(string mensaje)
        {
            // Leer una línea de la consola
            double numero;
            // Mientras no se pueda convertir a double o sea menor o igual a 0, seguir pidiendo el número
            while (true)
            {
                Console.WriteLine(mensaje);
                if (double.TryParse(Console.ReadLine(), out numero) && numero > 0)
                {
                    return numero;
                }
                Console.WriteLine("El número debe ser mayor que 0. Intente de nuevo:");
            }
        }
        // Leer un número entero
        public static int ReadInt()
        {
            // Leer una línea de la consola
            int numero;
            // Mientras no se pueda convertir a int, seguir pidiendo el número
            while (!int.TryParse(Console.ReadLine(), out numero))
            {
                Console.WriteLine("Entrada inválida. Intente de nuevo:");
            }
            return numero;
        }
        // Leer un texto
        public static string ReadText()
        {
            // Leer una línea de la consola
            string texto;
            // Mientras el texto esté vacío, seguir pidiendo el texto
            do
            {
                texto = Console.ReadLine().Trim();
                if (string.IsNullOrEmpty(texto))
                {
                    Console.WriteLine("El nombre no puede estar vacío. Intente de nuevo:");
                }
            } while (string.IsNullOrEmpty(texto));

            return texto;
        }
    }
}
